### PR TITLE
Expose ClientOptions and pipeline positioning in Form Recognizer builders

### PR DESCRIPTION
**Pull request by samvaity**
This PR 
- adds support to specify application Id using Client options
- updates client builders to leverage the support for pipeline positioning introduced in core. Ref - #15630

Closes: #15632, #15125

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerClientBuilder.java [hunk 1]
```diff
@@ -175,17 +180,28 @@ public FormRecognizerAsyncClient buildAsyncClient() {
     }
 
     private HttpPipeline getDefaultHttpPipeline(Configuration buildConfiguration) {
+        ClientOptions buildClientOptions = (clientOptions == null) ? new ClientOptions() : clientOptions;
```
Comment (srnagar) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerClientBuilder.java:183]: ClientOptions also allows users to add new headers. We should update our AddHeadersPolicy to include headers in ClientOptions.

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingClientBuilder.java [hunk 2]
```diff
@@ -176,21 +181,30 @@ private HttpPipeline getDefaultHttpPipeline() {
         // Global Env configuration store
         final Configuration buildConfiguration = (configuration == null)
             ? Configuration.getGlobalConfiguration().clone() : configuration;
+        ClientOptions buildClientOptions = (clientOptions == null) ? new ClientOptions() : clientOptions;
```
Comment (srnagar) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingClientBuilder.java:184]: All the comments for FRClientBuilder apply to the training client builder as well.

**Comment by samvaity** [comment]
Closed as work continued here https://github.com/Azure/azure-sdk-for-java/pull/16613

**Pull request closed without merging**